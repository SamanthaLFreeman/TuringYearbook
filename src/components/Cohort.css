.staff,
.students {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

h2 {
  background-color: rgba(255, 255, 255, 0.95);
  display: block;
  padding: 10px;
  color: blue;
}
