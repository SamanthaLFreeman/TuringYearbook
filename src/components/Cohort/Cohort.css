.staff,
.students {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

h2 {
  background-color: rgba(255, 255, 255, 0.95);
  display: block;
  font-family: "Lexend Zetta", sans-serif;
  padding: 10px;
  color: #035506;
}
