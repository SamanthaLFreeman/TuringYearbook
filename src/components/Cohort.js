import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = (props) => {
  const staff = props.staff.map(person => <Person individual={person} key={person.id} />)
  const students = props.students.map(person => <Person individual={person} key={person.id} />)

  return (
    <main className="cohortContainer">
      {staff}
      {students}
    </main>
  )
}

export default Cohort;
