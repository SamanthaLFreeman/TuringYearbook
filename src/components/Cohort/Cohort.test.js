import React from 'react';
import { shallow } from 'enzyme';
import Cohort from './Cohort';

describe('Cohort', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Cohort staff={[{
      id: 1,
      name: 'Robbie',
      quote: 'pants...', superlative: 'Most likely to',
      studentCheck: 'staff'
    }]} students={[{
      id: 1,
      name: 'Sam',
      quote: 'Huh?', superlative: 'Most likely to',
      studentCheck: 'student'
    }]} removeStudent={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
  })
})