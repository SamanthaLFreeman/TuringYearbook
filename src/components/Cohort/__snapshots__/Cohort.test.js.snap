// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot with all data passed in correctly 1`] = `
<main
  className="cohortContainer"
>
  <h2>
    Staff
  </h2>
  <div
    className="staff"
  >
    <Person
      individual={
        Object {
          "id": 1,
          "name": "Robbie",
          "quote": "pants...",
          "studentCheck": "staff",
          "superlative": "Most likely to",
        }
      }
      key="1"
      studentCheck="staff"
    />
  </div>
  <h2>
    Students
  </h2>
  <div
    className="students"
  >
    <Person
      individual={
        Object {
          "id": 1,
          "name": "Sam",
          "quote": "Huh?",
          "studentCheck": "student",
          "superlative": "Most likely to",
        }
      }
      key="1"
      removeStudent={[MockFunction]}
      studentCheck="student"
    />
  </div>
</main>
`;
